test/functional/Valhalla/src_lw5/org/openj9/test/lworld/ValueTypeTests.java

```diff
@@ -1362,23 +1387,414 @@ static public void testGCFlattenedMegaObjectArray() throws Throwable {
+		}
+	}
 
+	@Test(priority=2)
+	static public void testSynchMethodsOnRefTypes() throws Throwable {
+		Class<?> refTypeClass = ValueTypeGenerator.generateRefClass("TestSynchMethodsOnRefTypes");
+		MethodHandle makeRef = lookup.findStatic(refTypeClass, "makeRef", MethodType.methodType(refTypeClass));
+		Object refType = makeRef.invoke();
+
+		MethodHandle syncMethod = lookup.findVirtual(refTypeClass, "synchronizedMethodReturnInt", MethodType.methodType(int.class));
+		MethodHandle staticSyncMethod = lookup.findStatic(refTypeClass, "staticSynchronizedMethodReturnInt", MethodType.methodType(int.class));
+
+		try {
+			syncMethod.invoke(refType);
+		} catch (IllegalMonitorStateException e) {
+			fail("should not throw exception. Synchronized static methods can be used with RefType");
+		}
 
-	static void checkEqualPoint2D(Point2D point, int[] position) throws Throwable {
-		assertEquals(point.x, position[0]);
-		assertEquals(point.y, position[1]);
+		try {
+			staticSyncMethod.invoke();
+		} catch (IllegalMonitorStateException e) {
+			fail("should not throw exception. Synchronized static methods can be used with RefType");
+		}
 	}
 
-	static void checkEqualFlattenedLine2D(FlattenedLine2D line, int[][] positions) throws Throwable {
-		checkEqualPoint2D(line.st, positions[0]);
-		checkEqualPoint2D(line.en, positions[1]);
+	@Test(priority=1)
+	static public void testMonitorExitWithRefType() throws Throwable {
+		int x = 1;
+		Object refType = (Object) x;
+
+		Class<?> testMonitorExitWithRefType = ValueTypeGenerator.generateRefClass("TestMonitorExitWithRefType");
+		MethodHandle monitorExitWithRefType = lookup.findStatic(testMonitorExitWithRefType, "testMonitorExitOnObject", MethodType.methodType(void.class, Object.class));
+		try {
+			monitorExitWithRefType.invoke(refType);
+			fail("should throw exception. MonitorExit doesn't have a matching MonitorEnter");
+		} catch (IllegalMonitorStateException e) {}
 	}
 
-	static void checkEqualTriangle2D(Triangle2D triangle, int[][][] positions) throws Throwable {
-		checkEqualFlattenedLine2D(triangle.v1, positions[0]);
-		checkEqualFlattenedLine2D(triangle.v2, positions[1]);
-		checkEqualFlattenedLine2D(triangle.v3, positions[2]);
+	@Test(priority=1)
+	static public void testMonitorEnterAndExitWithRefType() throws Throwable {
+		int x = 2;
+		Object refType = (Object) x;
+
+		Class<?> testMonitorEnterAndExitWithRefType = ValueTypeGenerator.generateRefClass("TestMonitorEnterAndExitWithRefType");
+		MethodHandle monitorEnterAndExitWithRefType = lookup.findStatic(testMonitorEnterAndExitWithRefType, "testMonitorEnterAndExitWithRefType", MethodType.methodType(void.class, Object.class));
+		try {
+			monitorEnterAndExitWithRefType.invoke(refType);
+		} catch (IllegalMonitorStateException e) {
+			fail("shouldn't throw exception. MonitorEnter and MonitorExit should be used with refType");
+		}
+	}
+
+	static class ClassWithValueTypeVolatile {
```

Comment:
The original [class](https://github.com/eclipse-openj9/openj9/blob/df7853e07e78c5b225fcfd4d7805057cbf850cef/test/functional/Valhalla/src_qtypes/org/openj9/test/lworld/ValueTypeTests.java#L1454) is a value class. Is there a reason to change it to a non-value class ?

Reply:
nvm, I see the reason in the description: "modified only to test volatile value types in an identity class. volatile fields cannot exist in a value type since fields are implicitly final."